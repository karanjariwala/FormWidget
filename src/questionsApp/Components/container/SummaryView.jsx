import React from 'react';
import styled from 'styled-components';

import { connect } from 'react-redux';
import { getAnsweredQuestions } from './selector';
import SummaryCard from '../presentational/SummaryCard';

const Title = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

const SummaryView = ({ answeredQuestions }) => {
  let toRender = <span />;
  if (Array.isArray(answeredQuestions)) {
    toRender = (
      <div>
        <Title>Claim Submission</Title>
        {answeredQuestions.map(question => {
          return <SummaryCard key={question.id} {...question} />;
        })}
      </div>
    );
  }

  return toRender;
};

const mapStateToProps = state => ({
  answeredQuestions: getAnsweredQuestions(state),
});

export default connect(mapStateToProps)(SummaryView);
