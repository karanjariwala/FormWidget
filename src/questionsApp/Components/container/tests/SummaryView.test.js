/**
 * @jest-environment node
 */

import React from 'react';
import { unWrapped } from '../SummaryView.jsx';
import renderer from 'react-test-renderer';
it('renders correctly', () => {
  const props = {
    answeredQuestions: [
      {
        id: 'C02',
        text: 'Please provide us with a contact number?',
        reply: '',
        type: 'string',
        next: 'C03',
      },
      {
        id: 'C03',
        text: 'Please provide us with a contact email address?',
        reply: '',
        type: 'string',
        next: null,
      },
    ],
  };
  const tree = renderer.create(<unWrapped props={props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
