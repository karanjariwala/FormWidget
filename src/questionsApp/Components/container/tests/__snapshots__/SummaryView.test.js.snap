// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<unWrapped
  props={
    Object {
      "answeredQuestions": Array [
        Object {
          "id": "C02",
          "next": "C03",
          "reply": "",
          "text": "Please provide us with a contact number?",
          "type": "string",
        },
        Object {
          "id": "C03",
          "next": null,
          "reply": "",
          "text": "Please provide us with a contact email address?",
          "type": "string",
        },
      ],
    }
  }
/>
`;
